src/VisualStudio/Core/Def/Implementation/FindReferences/Entries/DocumentSpanEntry.cs

```diff
@@ -69,7 +69,12 @@ protected override IList<System.Windows.Documents.Inline> CreateLineTextInlines(
                                           .GetProperties(propertyId);
                 var highlightBrush = properties["Background"] as Brush;
 
-                var classifiedSpans = _excerptResult.ClassifiedSpans;
+                // Remove additive classified spans before creating classified text.
+                // Otherwise the text will be repeated since there are two classifications
+                // for the same span. Additive classifications should not change the foreground
+                // color, so the resulting classified text will retain the proper look.
```

Comment:
> Additive classifications should not change the foreground
> color, so the resulting classified text will retain the proper look.

but won't they do things like bold/italicize?   

it's probably fine to remove those.  but it might be worth calling out that we will lose that information.  

---

I only bring this up because the intent of the find-refs code is to make it look like an actual line from the buffer.  this is how we simulate it.  But these sorts of differences cause us to move further away.

Reply:
Cyrus, going to take this PR to get around this bug while I work on a more complete solution. Will track that work in https://github.com/dotnet/roslyn/issues/32770